sentry/src/middleware/auth.rs

```diff
@@ -0,0 +1,95 @@
+use hyper::header::AUTHORIZATION;
+use hyper::{Body, Request};
+use redis::aio::SharedConnection;
+
+use primitives::adapter::{Adapter, Session as AdapterSession};
+
+use crate::{ResponseError, Session};
+
+/// Check `Authorization` header for `Bearer` scheme with `Adapter::session_from_token`.
+/// If the `Adapter` fails to create an `AdapterSession`, `ResponseError::BadRequest` will be returned.
+pub(crate) async fn for_request(
+    mut req: Request<Body>,
+    adapter: impl Adapter,
+    redis: SharedConnection,
+) -> Result<Request<Body>, ResponseError> {
+    let authorization = req.headers().get(AUTHORIZATION);
+
+    let prefix = "Bearer ";
+
+    let token = authorization
+        .and_then(|hv| {
+            hv.to_str()
+                .map(|token_str| {
+                    if token_str.starts_with(prefix) {
+                        None
+                    } else {
+                        Some(token_str.replacen(prefix, "", 1))
+                    }
+                })
+                .transpose()
+        })
+        .transpose()?;
+
+    if let Some(ref token) = token {
+        let adapter_session = match redis::cmd("GET")
+            .arg(token)
+            .query_async::<_, Option<String>>(&mut redis.clone())
```

Comment:
AFAIK there is no deserializatioin/serialization for Redis yet, only this (serialization only):
https://github.com/OneSignal/serde-redis